Add metaplex token util

cotemper opened the pull request:
### Description

- Upgraded anchor version to v0.25.0.
- Add metadata token util functions.

cotemper committed:
add metaplex token util

long-banana commented on package.json:
```diff
@@ -41,7 +41,8 @@
   "dependencies": {
-    "@project-serum/anchor": "^0.24.2",
+    "@metaplex-foundation/mpl-token-metadata": "^2.2.4",
+    "@project-serum/anchor": "^0.25.0",
```
I'm hesitant to bump to this version until raindrops has some more automated testing in place.

cotemper replied:
I don't want to upgrade anchor version because the v0.24.2 doesn't have associated token program in anchor spl.
If we keep using anchor v0.24.2, we need to install extra spl-token package to support associated token program.

cotemper replied:
I will keep the v0.24.2 version then.

cotemper replied:
Hmm, actually don't need to add `@solana/spl-token` as an extra package in `package.json` because metaplex package includes it. Go to keep using anchor v0.24.2 then.

long-banana reviewed (changes requested):
Move this into a `token` directory instead of `util`.

cotemper committed:
update anchor version to v0.24.2

long-banana reviewed (changes requested):
```
src
> connection
> state
> token
> transaction
> wallet
```

no util at all

cotemper committed:
refactor token dir

cotemper commented:
> ```
> src
> > connection
> > state
> > token
> > transaction
> > wallet
> ```
> 
> no util at all

Removed the util directory and moved it to the token directory.

long-banana commented on src/index.ts:
```diff
@@ -4,3 +4,4 @@ export * as Transaction from './transaction';
 export * as Connection from './connection';
 export * as State from './state';
 export * as Wallet from './wallet';
+export * as Util from './token';
```
```suggestion
export * as Token from './token';
```

cotemper committed:
Update src/index.ts

Co-authored-by: Long Banana <100438324+long-banana@users.noreply.github.com>
